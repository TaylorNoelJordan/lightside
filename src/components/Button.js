import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

import './Button.scss';


const Button = (props) => {
    const {link, title, img, animateButtons} = props;

    return (
          <Link to={link}>
            <button className='selectCategoryBtn' onClick={() => animateButtons()}>
              <span className='selectCategoryBtnText'>{title}</span>
              <img className='icon' src={img} alt='' />
            </button>
          </Link>
      )
}

export default Button;