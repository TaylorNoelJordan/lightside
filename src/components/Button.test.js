import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button'


describe('Button', () => {
    let wrapper;
    let mockAnimateButtons;

    beforeEach(() => {
        mockAnimateButtons = jest.fn();
        wrapper = shallow(<Button 
            animateButtons={mockAnimateButtons} link='/people' title='people' img='img' /> )
    });

    it('should render all props passed down from ButtonContainer', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should update state when animateButtons is called', () => {
        wrapper.instance().animateButtons()

        expect(wrapper.state('selected')).toEqual(true);
    });

})