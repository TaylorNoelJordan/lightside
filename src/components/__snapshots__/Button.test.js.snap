// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render all props passed down from ButtonContainer 1`] = `
<Link
  to="/people"
>
  <button
    className="selectCategoryBtn"
    onClick={[Function]}
  >
    <span
      className="selectCategoryBtnText"
    >
      people
    </span>
    <img
      alt=""
      className="icon"
      src="img"
    />
  </button>
</Link>
`;
